src/anyp/Uri.h

```diff
@@ -57,6 +59,18 @@ class Uri
         port_ = std::nullopt;
         touch();
     }
+
+    void swap(Uri& other)
+    {
+        std::swap(scheme_, other.scheme_);
+        std::swap(userInfo_, other.userInfo_);
+        memcpy(host_, other.host_, sizeof(host_));
```

Comment:
is host_ not swapped?

Reply:
Yes, using ```memcpy``` with overlapping sequences is UB, but I think that
```Obj a = a;```
is pretty rare things although that's valid and compilable code.
I think that usage of ```memmove``` is good idea for handling this use case

Reply:
Please focus on the `= default` solution first. Using memmove() is unnecessary if that `= default` solution works.

Reply:
Sure